pyk/src/pyk/klean/__main__.py

```diff
@@ -0,0 +1,108 @@
+from __future__ import annotations
+
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from pyk.cli.utils import dir_path
+from pyk.kore.internal import KoreDefn
+
+if TYPE_CHECKING:
+    from argparse import Namespace
+    from collections.abc import Iterable
+
+
+def main() -> None:
+    import logging
+    import sys
+
+    from pyk.cli.utils import LOG_FORMAT
+
+    args = sys.argv
+    level, args = _extract_log_level(args)
+    logging.basicConfig(level=level, format=LOG_FORMAT)
+
+    klean(args)
+
+
+def _extract_log_level(args: list[str]) -> tuple[int, list[str]]:
+    from pyk.cli.args import KCLIArgs
+    from pyk.cli.utils import loglevel
+
+    parser = KCLIArgs().logging_args
+    ns, rest = parser.parse_known_args(args)
+    level = loglevel(ns)
+    return level, rest
+
+
+def klean(args: Iterable[str]) -> None:
+    from .generate import generate
+
+    ns = _parse_args(args)
+    defn = _load_defn(ns.definition_dir)
+    defn = defn.filter_rewrites(ns.rules)
```

Comment:
Since `ns.rules` can be `None`, maybe add something like the following?

```suggestion
    if ns.rules:
        defn = defn.filter_rewrites(ns.rules)
```

Reply:
Good catch.